TAP testcase 'simple expires test' sometimes failed (flaky?)

https://github.com/tarantool/expirationd/runs/3910102548?check_suite_focus=true

```
TAP version 13
1..10
    # simple expires test
    1..4
    ok - checking start time
    ok - checking task name
    ok - checking restart count
    not ok - Test task executed and moved to archive
make: *** [Makefile:39: test] Error 1
      ---
      filename: ./test.lua
      trace:
      - line: 296
        source: '@./test.lua'
        filename: ./test.lua
        what: Lua
        namewhat: local
        name: fun
        src: ./test.lua
      - line: 218
        source: '@builtin/tap.lua'
        filename: builtin/tap.lua
        what: Lua
        namewhat: method
        name: test
        src: builtin/tap.lua
      - line: 0
        source: '@./test.lua'
        filename: ./test.lua
        what: main
        namewhat: 
        src: ./test.lua
      line: 0
      expected: 7
      got: 8
      ...
    # simple expires test: end
not ok - failed subtests
  ---
  filename: ./test.lua
  trace:
  - line: 245
    source: '@builtin/tap.lua'
    filename: builtin/tap.lua
    what: Lua
    namewhat: method
    name: test
    src: builtin/tap.lua
  - line: 0
    source: '@./test.lua'
    filename: ./test.lua
    what: main
    namewhat: 
    src: ./test.lua
  planned: 4
  failed: 1
  line: 0
  ...
    # execution error test
    1..2
    ok - checking restart count
    ok - Error task executed
    # execution error test: end
```

Tarantool version 2.8

How-To reproduce:

```
$ INDEX_TYPE='HASH' ./test.lua
```

--- a/test.lua
+++ b/test.lua
@@ -343,7 +343,14 @@ test:test('simple expires test',  function(test)
     test:is(task.name, "test", 'checking task name')
     local restarts = 1
     test:is(task.restarts, restarts, 'checking restart count')
-    test:is(task.expired_tuples_count, 7, 'Test task executed and moved to archive')
+    local res = wait_cond(
+        function()
+            local task = expirationd.task("test")
+            local cnt = task.expired_tuples_count
+            return cnt == 7
+        end
+    )
+    test:is(res, true, 'Test task executed and moved to archive')
     expirationd.kill("test")
 end)
 

